From 4fff3e33c39807680dab5961e6577fa128237921 Mon Sep 17 00:00:00 2001
From: Arlo Breault <arlolra@gmail.com>
Date: Sun, 28 Aug 2016 08:57:41 -0700
Subject: [PATCH 08/20] Bug 1298574 - Set _userVCard own property when
 downloading vCard fails. r=aleth

 * This prevents an infinite req / res cycle.
---
 chat/protocols/xmpp/xmpp.jsm | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/chat/protocols/xmpp/xmpp.jsm b/chat/protocols/xmpp/xmpp.jsm
index 4420a1d..71f7770 100644
--- a/chat/protocols/xmpp/xmpp.jsm
+++ b/chat/protocols/xmpp/xmpp.jsm
@@ -2243,6 +2243,20 @@ var XMPPAccountPrototype = {
                                  .replace(/.{74}/g, "$&\n");
       }
     }
+    else {
+      // Downloading the vCard failed.
+      if (this.handleErrors({
+          itemNotFound: () => false,  // OK, no vCard exists yet.
+          default: () => true
+        })(aStanza)) {
+        this.WARN("Unexpected error retrieving the user's vcard, " +
+          "so we won't attempt to set it either.");
+        return;
+      }
+      // Set this so that we don't get into an infinite loop trying to download
+      // the vcard again. The check in sendVCard is for hasOwnProperty.
+      this._userVCard = null;
+    }
     this._sendVCard();
   },
 
-- 
2.10.1

